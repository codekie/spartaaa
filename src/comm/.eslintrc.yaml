extends:
  - ../../.eslintrc.yaml
rules:
  new-cap: off
