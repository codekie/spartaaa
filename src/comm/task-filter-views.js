export const pending = 'pending';
export const all = 'all';
