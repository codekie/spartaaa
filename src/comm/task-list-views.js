export const all = 'all';
export const completed = 'completed';
export const next = 'next';
export const pending = 'pending';
