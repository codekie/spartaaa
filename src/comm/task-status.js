export default {
    pending: 'pending',
    deleted: 'deleted'
};
