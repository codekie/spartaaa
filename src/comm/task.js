import { Record, List } from 'immutable';

const EMPTY_TASK = Record({
    description: null,
    due: null,
    entry: null,
    id: null,
    mask: null,
    modified: null,
    priority: null,
    project: null,
    recur: null,
    start: null,
    status: null,
    tags: List(),
    urgency: null,
    uuid: null,
    wait: null
})();

// Merging the data with a base-immutable, will automatically map `Array`s to `List`s, in contrary to the
// generated record-factory
export default (data = {}) => EMPTY_TASK.merge(data);
export {
    EMPTY_TASK
};
