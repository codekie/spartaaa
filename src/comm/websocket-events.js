// CONSTANTS

const REQUEST = 'Request',
    RESPONSE = 'Response',
    SESSION = 'session',
    SESSION__UPDATE = `${ SESSION }.update`,
    TASKS = 'tasks',
    TASKS__GET = `${ TASKS }.get`,
    TASKS__ACTIVATE = `${ TASKS }.activate`,
    TASKS__DEACTIVATE = `${ TASKS }.deactivate`;

// PUBLIC API

export {
    getRequestEventName,
    getResponseEventName
};
export const Event = {
    session: {
        update: SESSION__UPDATE
    },
    tasks: {
        default: TASKS,
        get: TASKS__GET,
        activateTask: TASKS__ACTIVATE,
        deactivateTask: TASKS__DEACTIVATE
    }
};

// IMPLEMENTATION DETAILS

// Public

function getRequestEventName(eventNode) {
    return `${ REQUEST }.${ _getEventName(eventNode) }`;
}

function getResponseEventName(eventNode) {
    return `${ RESPONSE }.${ _getEventName(eventNode) }`;
}

// Private

function _getEventName(eventNode) {
    return typeof eventNode === 'string' ? eventNode : eventNode.default;
}
