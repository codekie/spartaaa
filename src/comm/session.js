import * as VIEW_NAME from './task-list-views';
import TaskFilter from './session/task-filter';

export default class Session {
    constructor() {
        this.taskFilter = new TaskFilter();
        this.viewName = VIEW_NAME.pending;
    }
}
