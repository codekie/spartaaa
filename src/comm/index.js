import * as WebSocketEvents from './websocket-events';

export {
    WebSocketEvents
};
