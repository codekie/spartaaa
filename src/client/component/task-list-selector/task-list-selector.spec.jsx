import React from 'react';
import TaskListSelector from './task-list-selector.jsx';
import { shallow } from 'enzyme';
import * as TaskListView from '../../../comm/task-list-views';

describe('renders correctly', () => {
    Object.keys(TaskListView)
        .map(key => TaskListView[key])
        .forEach((viewName) => {
            describe(`\`${ viewName }\`-task-list-selector`, () => {
                let _elmt = null,
                    _context = null;

                // eslint-disable-next-line max-nested-callbacks
                beforeEach(() => {
                    _context = _createContext();
                    _elmt = _context.createElement({ viewName });
                });

                // eslint-disable-next-line max-nested-callbacks
                it('should render the element', () => expect(_elmt).toMatchSnapshot());
                // eslint-disable-next-line max-nested-callbacks
                it('should call the click-handler with the correct value', () => {
                    _elmt.find(`.view-tab-${ viewName }`).simulate('click');
                    expect(_context.setTaskListViewAndUpdateList.mock.calls.length).toBe(1);
                });
            });
        });
});

function _createContext() {
    const setTaskListViewAndUpdateList = jest.fn();
    return {
        createElement: ({ viewName }) => shallow(<TaskListSelector
            setTaskListViewAndUpdateList={setTaskListViewAndUpdateList} viewName={viewName} />),
        setTaskListViewAndUpdateList
    };
}
