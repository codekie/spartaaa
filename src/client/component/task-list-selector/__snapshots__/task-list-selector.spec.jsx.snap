// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly \`all\`-task-list-selector should render the element 1`] = `
<ForwardRef
  className="panel-tabs"
  renderAs="div"
>
  <ForwardRef
    active={false}
    className="view-tab-pending"
    onClick={[Function]}
    renderAs="a"
  >
    Pending
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-next"
    onClick={[Function]}
    renderAs="a"
  >
    Next
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-postponed"
    onClick={[Function]}
    renderAs="a"
  >
    Postponed
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-completed"
    onClick={[Function]}
    renderAs="a"
  >
    Completed
  </ForwardRef>
  <ForwardRef
    active={true}
    className="view-tab-all"
    onClick={[Function]}
    renderAs="a"
  >
    All
  </ForwardRef>
</ForwardRef>
`;

exports[`renders correctly \`completed\`-task-list-selector should render the element 1`] = `
<ForwardRef
  className="panel-tabs"
  renderAs="div"
>
  <ForwardRef
    active={false}
    className="view-tab-pending"
    onClick={[Function]}
    renderAs="a"
  >
    Pending
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-next"
    onClick={[Function]}
    renderAs="a"
  >
    Next
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-postponed"
    onClick={[Function]}
    renderAs="a"
  >
    Postponed
  </ForwardRef>
  <ForwardRef
    active={true}
    className="view-tab-completed"
    onClick={[Function]}
    renderAs="a"
  >
    Completed
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-all"
    onClick={[Function]}
    renderAs="a"
  >
    All
  </ForwardRef>
</ForwardRef>
`;

exports[`renders correctly \`next\`-task-list-selector should render the element 1`] = `
<ForwardRef
  className="panel-tabs"
  renderAs="div"
>
  <ForwardRef
    active={false}
    className="view-tab-pending"
    onClick={[Function]}
    renderAs="a"
  >
    Pending
  </ForwardRef>
  <ForwardRef
    active={true}
    className="view-tab-next"
    onClick={[Function]}
    renderAs="a"
  >
    Next
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-postponed"
    onClick={[Function]}
    renderAs="a"
  >
    Postponed
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-completed"
    onClick={[Function]}
    renderAs="a"
  >
    Completed
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-all"
    onClick={[Function]}
    renderAs="a"
  >
    All
  </ForwardRef>
</ForwardRef>
`;

exports[`renders correctly \`pending\`-task-list-selector should render the element 1`] = `
<ForwardRef
  className="panel-tabs"
  renderAs="div"
>
  <ForwardRef
    active={true}
    className="view-tab-pending"
    onClick={[Function]}
    renderAs="a"
  >
    Pending
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-next"
    onClick={[Function]}
    renderAs="a"
  >
    Next
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-postponed"
    onClick={[Function]}
    renderAs="a"
  >
    Postponed
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-completed"
    onClick={[Function]}
    renderAs="a"
  >
    Completed
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-all"
    onClick={[Function]}
    renderAs="a"
  >
    All
  </ForwardRef>
</ForwardRef>
`;

exports[`renders correctly \`postponed\`-task-list-selector should render the element 1`] = `
<ForwardRef
  className="panel-tabs"
  renderAs="div"
>
  <ForwardRef
    active={false}
    className="view-tab-pending"
    onClick={[Function]}
    renderAs="a"
  >
    Pending
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-next"
    onClick={[Function]}
    renderAs="a"
  >
    Next
  </ForwardRef>
  <ForwardRef
    active={true}
    className="view-tab-postponed"
    onClick={[Function]}
    renderAs="a"
  >
    Postponed
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-completed"
    onClick={[Function]}
    renderAs="a"
  >
    Completed
  </ForwardRef>
  <ForwardRef
    active={false}
    className="view-tab-all"
    onClick={[Function]}
    renderAs="a"
  >
    All
  </ForwardRef>
</ForwardRef>
`;
